### ISSUE TITLE
callkeep example with any RTC/VOIP like agora.io

### ISSUE THREAD

**Issue by viztushar** [flutter-webrtc/callkeep#101]
please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works. 

thanks.

**Comment by hmarat** [comment]
> please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works.
> 
> thanks.

Have you implemented it?

**Comment by devDonald** [comment]
> please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works.
> 
> thanks.

Please Have you been able to implement this

**Comment by viztushar** [comment]
> > please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works.
> 
> > 
> 
> > thanks.
> 
> 
> 
> Please Have you been able to implement this

Yes, maybe it's not right way but I have directly implemented the agora code into CallKeep.

**Comment by devDonald** [comment]
> > > please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works.
> > 
> > 
> > > 
> > 
> > 
> > > thanks.
> > 
> > 
> > Please Have you been able to implement this
> 
> Yes, maybe it's not right way but I have directly implemented the agora code into CallKeep.

Please, can you guide me on how to do it. Or maybe some code sample

**Comment by benyaminbeyzaie** [comment]
> > > please provide Callkeep example with any RTC/VOIP like agora.io. so that we can understand much batter how the callkeep implement works.
> > 
> > 
> > > 
> > 
> > 
> > > thanks.
> > 
> > 
> > Please Have you been able to implement this
> 
> Yes, maybe it's not right way but I have directly implemented the agora code into CallKeep.

Can you please share your solution? Thanks

**Comment by viztushar** [comment]
@devDonald @benyaminbeyzaie 

as i said this is maybe not right way but it get job done

first declare agora variable(callkeep.h) and in callkeep.pod file
![Screenshot 2021-09-22 at 2 44 17 PM](https://user-images.githubusercontent.com/6755833/134317132-e04a8228-111a-48b1-bb15-4ed072c8831d.png)

(callkeep.m)
step 1: - initialize agora with key 
![Screenshot 2021-09-22 at 2 43 55 PM](https://user-images.githubusercontent.com/6755833/134317291-3a357b80-feee-4c6a-bff5-6c8e65f09d59.png)

step 2: get data from firebase or voip push and save to the NSUserDefaults in ios and shared preferences in android using method channel

![Screenshot 2021-09-22 at 2 38 39 PM](https://user-images.githubusercontent.com/6755833/134317537-f797bc66-3e29-436e-84a1-fa948912e8f9.png)

step 3: create one more method if you want to use those firebase and voip push data into flutter called "getData" this will return the all the save data(above)

![Screenshot 2021-09-22 at 2 38 54 PM](https://user-images.githubusercontent.com/6755833/134317850-2540f3f6-e8ae-4b21-814d-7ea81867244a.png)

step 4: if user answer the call then init data with agora

![Screenshot 2021-09-22 at 2 40 01 PM](https://user-images.githubusercontent.com/6755833/134318625-27ca6c17-8871-4f53-abc4-c5f52d30b6d6.png)

end call
![Screenshot 2021-09-22 at 2 40 09 PM](https://user-images.githubusercontent.com/6755833/134318740-3f295fef-a195-4bed-bb13-6469f2733e28.png)

and if user leave the channel
![Screenshot 2021-09-22 at 2 40 45 PM](https://user-images.githubusercontent.com/6755833/134318810-15082511-a795-4b4c-93a7-2de8f9d07eb7.png)

**Comment by bihim** [comment]
Are these swift code or objective C?

**Comment by viztushar** [comment]
> Are these swift code or objective C?

Objective c
